### PR TITLE
fix: Replace .platform-helper-version check in buildspec

### PR DESCRIPTION
**Pull request by antroy-madetech**
Addresses [DBTP-1153](https://uktrade.atlassian.net/browse/DBTP-1153)

Buildspec changes to support the move away from .platform-helper-version.

Also makes the `platform-helper version get-platform-helper-for-project` command fail if the version is not in the platform-config.yml and there isn't a .platform-helper-version file with a version to make it easier to use in pipelines.

---
### Title:
- [x] Scope included as per [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] Ticket reference included (unless it's a quick out of ticket thing)
### Description:
- [x] Link to ticket included (unless it's a quick out of ticket thing)
- [x] Includes tests (or an explanation for why it doesn't)
- [x] If the work includes user interface changes, before and after screenshots included in description
- [x] Includes any applicable changes to the documentation in this code base
- [x] Includes link(s) to any applicable changes to the documentation in the [DBT Platform Documentation](https://platform.readme.trade.gov.uk/) (can be to a pull request)
### Tasks:
- [ ] [Trigger the pull request regression tests for this branch](https://github.com/uktrade/platform-tools?tab=readme-ov-file#regression-tests) and confirm that they are passing

[DBTP-1153]: https://uktrade.atlassian.net/browse/DBTP-1153?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: dbt_platform_helper/templates/pipelines/environments/buildspec.yml [hunk 1]
```diff
@@ -14,29 +14,31 @@ phases:
           echo "Cannot find .copilot-version file"
           exit 1
         fi
+      - COPILOT_VERSION=$(cat .copilot-version)
+      # Install pyyaml and dbt-platform-helper
+      - pip install PyYAML dbt-platform-helper
```
Comment (gabelton) [inline dbt_platform_helper/templates/pipelines/environments/buildspec.yml:19]: Isn't PyYAML already installed with dbt-platform-helper?

**Review by gabelton — APPROVED** [review]

File: dbt_platform_helper/templates/pipelines/environments/buildspec.yml (hunk 1 above)
Comment (gabelton) [reply dbt_platform_helper/templates/pipelines/environments/buildspec.yml:19]: Hadn't noticed this gets used in post-build. Happy to leave it in there

File: dbt_platform_helper/utils/versioning.py [hunk 2]
```diff
@@ -245,7 +245,9 @@ def check_platform_helper_version_mismatch():
```
Comment (antroy-madetech) [inline dbt_platform_helper/utils/versioning.py:245]: @WillGibson Do you know why this check is here? It seems to work fine when running locally with this check commented out. It means there's some functionality which can't be tested manually before releasing which isn't ideal.

**Review by gabelton — APPROVED** [review]

File: dbt_platform_helper/utils/versioning.py (hunk 2 above)
Comment (WillGibson) [reply dbt_platform_helper/utils/versioning.py:245]: It is to stop it messing up when you are using a locally built version of `platform-helper`, not one installed from PyPi.

E.g. in the regression tests.

Comment (antroy-madetech) [reply dbt_platform_helper/utils/versioning.py:245]: I added a test around this check.

**Pull request merged**